Add emit_json_defaults option

tobiaskohlbau opened the pull request:
The package jsonpb which is used to un/marshal json values
into protobuf structures has the option to emit default values
within generated json structures. This CL adds support for
this feature by adding emit_json_defaults as command line flag.
Use for e.g. with `--twirp_out=emit_json_defaults=ENABLE:.`.
Notice that once a value is set this feature is turned on. To
disable it just omit `emit_json_defaults` like for e.g.
`--twirp_out=.`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

larrymyers commented:
Looks like this would provide a way forward for my request in #103.

lvht commented:
👍 I need these feature as well. However, as the twirp does not use go mod, I think @tobiaskohlbau should remove the `go.mod` and `go.sum` to make this PR accepted.

lvht commented:
any maintainer is here?

pkieltyka commented:
pretty key feature. 0 is a real value. thank you for adding this to the generator

tobiaskohlbau commented:
@lvht thanks for pointing out the accidental added go modules files, I've removed them and rebased the commit.

Hope some maintainer has a point in the future if this could be merged or not.

marioizquierdo commented:
Unfortunately, this will have to stay in a branch for now. But there may be a possibility of adding this feature to a major new release (e.g. version 6 or 7). This may potentially break things, but it may be useful.

Lets keep the discussion in the issue https://github.com/twitchtv/twirp/issues/103. I added a long comment exploring different points of view: https://github.com/twitchtv/twirp/issues/103#issuecomment-451582978

dpolansky commented:
Closing this for now. See the discussion in #103 for more context.
